src/Account.mo

```diff
@@ -0,0 +1,54 @@
+import Result "mo:base/Result";
+
+import Player "Player";
+
+module {
+    public type Score = (AccountId, Nat);
+
+    public type AccountId = Nat;
+
+    public type Account = {
+        alias         : ?Text;
+        avatar        : ?Text;
+        flavorText    : ?Text;
+        id            : AccountId;
+        plugAddress   : ?Principal;
+        primaryWallet : Player.Player;
+        stoicAddress  : ?Principal;
+    };
+
+    // Represents the public API of the Metascore canister to authenticate users.
+    public type PublicInterface = actor {
+        // Returns the account associated with the given identifier.
+        getAccount          : query  (AccountId)               -> async Result.Result<Account, ()>;
+        // Updates an existing account.
+        updateAccount       : shared (UpdateRequest)           -> async UpdateResponse;
+        // Authentication of an account.
+        authenticateAccount : shared (AuthenticationRequest)   -> async AuthenticationResponse;
+    };
+
+    // Account authentication request.
+    public type AuthenticationRequest = {
+        #authenticate : Player.Player;
+        #link         : (Player.Player, Player.Player);
```

Comment:
We can rely on `shared ({ caller })` for one of the principals here. We don't need include params for both.

Reply:
I guess not actually, because we need to know if it's stoic or plug. YOU RIGHT!